src/basic/terminal-util.c

```diff
@@ -2559,6 +2575,40 @@ int terminal_get_terminfo_by_dcs(
         return r < 0 ? r : 0;
 }
 
+const char* default_term_for_tty(const char *tty) {
+        return tty && tty_is_vc_resolve(tty) ? "linux" : "vt220";
+}
+
+int query_term_for_tty(const char *tty, char **ret_term) {
+        _cleanup_free_ char *dcs_term = NULL;
+        int r;
+
+        assert(tty);
+        assert(ret_term);
+
+        /* Try to query the terminal implementation that we're on. This will not work in all
+         * cases, which is fine, since this is intended to be used as a fallback. */
+
+        _cleanup_close_ int tty_fd = open_terminal(tty, O_RDWR|O_NOCTTY|O_CLOEXEC);
+        if (tty_fd < 0)
+                return log_debug_errno(tty_fd, "Failed to open %s to query terminfo: %m", tty);
+
+        r = terminal_get_terminfo_by_dcs(tty_fd, tty_fd, &dcs_term);
```

Comment:
does the linux VT respond to this?

i think if we recognize the tty string as linux VT, we should return "linux" right away, without any expensive checking

Reply:
We might also want to exclude PTYs, but I don't have strong opinion about that. Can be adjusted later if needed.

Reply:
a pty can be anything. If we don't do the dcs stuff there there's no point in it at all.

Reply:
I expressed my viewpoint for the ptys [here](https://github.com/systemd/systemd/issues/36994#issuecomment-2895320937):

> Ptys can spawn over a network e.g. by SSH, in which case $TERM is handled by the protocol itself, and for locally spawned ptys by GUI terminals there is no issue either, so let's keep them apart for now.

Let's continue the discussion in that thread because it shouldn't block this patch.

Reply:
> does the linux VT respond to this?

Doesn't seem so.

> i think if we recognize the tty string as linux VT, we should return "linux" right away, without any expensive checking

OK, I'll update like that.